tests/auto/layouts/tst_layouts.cpp

```diff
@@ -51,27 +57,28 @@ tst_Layouts::tst_Layouts()
 
 void tst_Layouts::hBoxLayout()
 {
-    HBoxLayoutWidget* uiWidget = new HBoxLayoutWidget();
-
-    DeclarativeWidgetsDocument* document = new DeclarativeWidgetsDocument(QStringLiteral("qrc:/qml/HBoxLayoutTest.qml"), this);
-    QWidget* declarativeWidget = document->create<QWidget>();
-
-    QVERIFY2(declarativeWidget != nullptr, "Failed to create widget from document");
+    testLayouts( QWidgetPtr(new HBoxLayoutWidget())
+                 , DeclarativeWidgetsDocumentPtr(new DeclarativeWidgetsDocument(QStringLiteral("qrc:/qml/HBoxLayoutTest.qml"))));
+}
 
-    // Show the widgets to trigger geometry updates
-    uiWidget->show();
-    declarativeWidget->show();
+void tst_Layouts::vBoxLayout()
+{
+    testLayouts( QWidgetPtr(new VBoxLayoutWidget())
```

Comment:
Perform every explicit resource allocation on its own statement. This is unsafe. (Duplicates).